security/pkg/nodeagent/caclient/credentials.go

```diff
@@ -77,8 +78,8 @@ func (t *TokenProvider) RequireTransportSecurity() bool {
 // volatile memory), we can still proceed and allow other authentication methods to potentially
 // handle the request, such as mTLS.
 func (t *TokenProvider) GetToken() (string, error) {
-	if !t.forCA {
-		// When not using a CA, we only support reading from file
+	if !t.forCA && !t.opts.XdsEnableTokenExchange {
```

Comment:
The flag is called `EnableTokenExchange` but its actually enabling TokenExchange *and* credential fetcher, which may be confusing